testing/dart/lerp_test.dart

```diff
@@ -43,4 +43,48 @@ void main() {
     expect(lerpDouble(0.0, 10.0, 5.0), 50.0);
     expect(lerpDouble(10.0, 0.0, 5.0), -40.0);
   });
+
+  test('lerpDouble should return NaN if any input is NaN', () {
+    expect(lerpDouble(0.0, 10.0, double.nan), isNaN);
+    expect(lerpDouble(0.0, double.infinity, double.nan), isNaN);
+    expect(lerpDouble(0.0, double.nan, 5.0), isNaN);
+    expect(lerpDouble(0.0, double.nan, double.infinity), isNaN);
+    expect(lerpDouble(0.0, double.nan, double.nan), isNaN);
+    expect(lerpDouble(double.infinity, 10.0, double.nan), isNaN);
+    expect(lerpDouble(double.infinity, double.infinity, double.nan), isNaN);
+    expect(lerpDouble(double.infinity, double.nan, 5.0), isNaN);
+    expect(lerpDouble(double.infinity, double.nan, double.infinity), isNaN);
+    expect(lerpDouble(double.infinity, double.nan, double.nan), isNaN);
+    expect(lerpDouble(double.nan, 10.0, 5.0), isNaN);
+    expect(lerpDouble(double.nan, 10.0, double.infinity), isNaN);
+    expect(lerpDouble(double.nan, 10.0, double.nan), isNaN);
+    expect(lerpDouble(double.nan, double.infinity, 5.0), isNaN);
+    expect(lerpDouble(double.nan, double.infinity, double.infinity), isNaN);
+    expect(lerpDouble(double.nan, double.infinity, double.nan), isNaN);
+    expect(lerpDouble(double.nan, double.nan, 5.0), isNaN);
+    expect(lerpDouble(double.nan, double.nan, double.infinity), isNaN);
+    expect(lerpDouble(double.nan, double.nan, double.nan), isNaN);
+  });
+
+  test('lerpDouble returns NaN if interpolation results in Infinity - Infinity', () {
```

Comment:
I feel like there has to be a better way to word this, but can't think of anything great.